File: artguard-frontend/src/services/authService.js
import { signInWithPopup } from 'firebase/auth';
import { auth, provider } from './firebase';

import axios from 'axios';

const API_URL = 'http://127.0.0.1:3000/api'; 

export const login = async (email, password) => {
  const response = await axios.post(`${API_URL}/login`, {
    email,
    password,
  });
  return response.data; 
};

export const getMe = async (token) => {
  const response = await axios.get(`${API_URL}/me`, {
    headers: {
      Authorization: `Bearer ${token}`,
    },
  });
  return response.data;
};

export const loginWithGoogle = async () => {
  try {
    const result = await signInWithPopup(auth, provider);
    const idToken = await result.user.getIdToken();

    const response = await axios.post(`${API_URL}/google-login`, {
      idToken,
    });

    localStorage.setItem('token', response.data.token);
    return { success: true, token: response.data.token };

  } catch (error) {
    console.error("Google login error:", error);
    return {
      success: false,
      error: error.response?.data?.error || error.message,
    };
  }
};

File: ArtGuardMobile/app/src/main/java/com/example/artguardmobile/data/network/AuthService.kt
package com.example.artguardmobile.data.network

import retrofit2.Response
import retrofit2.http.Body
import retrofit2.http.POST

data class LoginRequest(val email: String, val password: String)
data class LoginResponse(val token: String)

data class RegisterRequest(
    val name: String,
    val email: String,
    val password: String
)
data class RegisterResponse(val token: String)

data class GoogleLoginRequest(val idToken: String)

interface AuthService {
    @POST("/api/login")
    suspend fun login(@Body request: LoginRequest): Response<LoginResponse>

    @POST("/api/register")
    suspend fun register(@Body request: RegisterRequest): Response<RegisterResponse>

    @POST("/api/google-login")
    suspend fun loginWithGoogle(@Body request: GoogleLoginRequest): Response<LoginResponse>
}


